doc/frontend.md

```diff
@@ -237,11 +244,75 @@ follow the existing pattern.
 `theme_changed {"name": "InspiredGitHub", "theme": Theme}`
 
 Notifies the client that the theme has been changed. The client should
-use the new theme to set colors as appropriate.
-The `Theme` object is directly serialized from a 
-[`syntect::highlighting::ThemeSettings`](https://github.com/trishume/syntect/blob/master/src/highlighting/theme.rs#L27) 
+use the new theme to set colors as appropriate. The `Theme` object is
+directly serialized from a [`syntect::highlighting::ThemeSettings`](https://github.com/trishume/syntect/blob/master/src/highlighting/theme.rs#L27)
 instance.
 
+### plugins
+
+#### available_plugins
+
+`available_plugins {"view_id": "view-id-1", "plugins": [{"name": "syntect",
+"running": true]}`
+
+Notifies the client of the plugins available to the given view.
+
+#### plugin_started
+
+`plugin_started {"view_id": "view-id-1", "plugin": "syntect"}`
+
+Notifies the client that the named plugin is running.
+
+#### plugin_stopped
+
+`plugin_stopped {"view_id": "view-id-1", "plugin": "syntect", "code" 101}`
+
+Notifies the client that the named plugin has stopped. The `code` field is an
+integer exit code; currently 0 indicates a user-initiated exit and 1 indicates
+an abnormal exit, i.e. a plugin crash.
+
+#### update_cmds
+
+`update_cmds {"view_id": "view-id-1", "plugin", "syntect", "cmds": [Command]}`
+
+Notifies the client of a change in the available commands for a given plugin.
+The `cmds` field is a list of all commands currently available to this plugin.
+Clients should store commands on a per-plugin basis; when the `cmds` argument is
+an empty list it means that this plugin is providing no commands; any previously
+available commands should be disabled.
+
+The format for describing a `Command` is in flux. The best place to look for
+a working example is in the tests in core-lib/src/plugins/manifest.rs. As of
+this writing, the following is valid json for a `Command` object:
+
+```json
+    {
+    "title": "Test Command",
+    "description": "Passes the current test",
+    "rpc_cmd": {
+        "type": "notification",
+        "method": "test.cmd",
+        "params": {
+            "this_works": "some value",
```

Comment:
My original thinking is that `params` is provided separately from `args`; in this case params could contain arguments that are not user supplied, which would be sent unchanged. So, for instance, a plugin could provide multiple commands that share a common rpc method. `args` are things we prompt the user for, and are used to update the `params` object.

I appreciate this definitely isn't clear, right now; and the examples don't help.

My goal with this plugins path work has been to get to a point where there's enough functionality and API in place for me to start actually writing _useful_ plugins; at that point I expect to iterate on a lot of these nitty things as I develop a better sense for the ergonomics, and at that point I'd also like to write up good documentation / examples (this is mentioned in #253). So fwiw I don't expect much of this code to be intact come september, and that belief has been reflected in the level of polish I've put into certain areas of it.